libs/gi/ui/src/components/build/CompareBuildWrapper.tsx

```diff
@@ -59,9 +59,19 @@ function CharacterWrapper({ artId, weaponId, onHide, onEquip }: WrapperProps) {
   } = useContext(CharacterContext)
   const newArt = database.arts.get(artId ?? '')
   const currentArtifactIds = equippedArtifacts
-  const newArtifactIds = objMap(currentArtifactIds, (art, slot) =>
-    slot === newArt?.slotKey ? artId : art
-  )
+  const newArtifactIds = objMap(currentArtifactIds, (art, slot) => {
```

Comment:
The fixed logic should be the same as above, in line 33, 
```suggestion
  const newArtifactIds = objMap(currentArtifactIds, (art, slot) =>
    slot === artId ? undefined : newArt?.slotKey === slot ? artId : art
  )
```